src/App/mermaidData/pullApiData.js

```diff
@@ -16,10 +16,28 @@ export const pullApiData = async ({
   dexiePerUserDataInstance,
   getAccessToken,
   projectId,
+  handleUserDeniedSyncPull,
 }) => {
-  if (!getAccessToken || !apiBaseUrl || !apiDataNamesToPull || !dexiePerUserDataInstance) {
+  if (
+    !apiBaseUrl ||
+    !apiDataNamesToPull ||
+    !dexiePerUserDataInstance ||
+    !getAccessToken ||
+    !handleUserDeniedSyncPull
+  ) {
     throw new Error('pullApiData is missing a required parameter')
   }
+
+  let shouldHandleUserDeniedSyncPull = false
+
+  let shouldDeleteProjectFromBeingConsideredOfflineReady = false
```

Comment:
a bit confused by this name - is offlineReady an array of projects?

`removeProjectFromOfflineReadyStatus` maybe?

Reply:
ya! `removeProjectFromOfflineReadyStatusList`. Im going to make that change with the coming very soon tests to save some 11th hour git bureaucracy if you dont mind

Reply:
but with the word should to indicate it is a boolean. `shouldRemoveProjectFromOfflineReadyStatusList`